nemo/utils/exp_manager.py

```diff
@@ -973,7 +973,7 @@ def configure_checkpointing(
 
 def check_slurm(trainer):
     try:
-        return trainer.accelerator_connector.is_slurm_managing_tasks
+        return int(os.getenv('SLURM_STEP_NUM_TASKS', '0')) == trainer.world_size
```

Comment:
Why is this being modified ? We want to depend on lightning in 2.0 to tell us if slurm is setup or not.

Reply:
Based on https://github.com/Lightning-AI/pytorch-lightning/pull/10353, seem that the `is_slurm_managing_tasks` attribute was dropped in 1.6.0; see also https://lightning.ai/docs/pytorch/1.6.0/generated/CHANGELOG.html#id5.

Reply:
@titu1994  Would this fix still need to set actimeo=0 config at NFS level for distributed checkpointing?

Reply:
@itzsimpl do you know if PTL provides a replacement for `is_slurm_managing_tasks` that we can use ? Or is the only to rely on `SLURM_STEP_NUM_TASKS`

Reply:
Based on my quick search through PTL and elsewhere in the last couple of days, I haven't found anything, hence the change. Note that here `check_slurm()` is based on the only case where I found it used in NeMo, i.e.:
https://github.com/NVIDIA/NeMo/blob/f6e64859174920e6a7fb26c4308cbe4c1d44206e/nemo/utils/exp_manager.py#L555-L559
 
It will return `True` only when SLURM creates the parallel tasks. It will not return `True` in the case where one uses SLURM to run an interactive job with multiple GPUs available, where PTL creates the distributed processes (https://lightning.ai/docs/pytorch/stable/clouds/cluster_advanced.html#interactive-mode).

There is also a note about the latter case in Lightning main branch
https://github.com/Lightning-AI/pytorch-lightning/blob/5a0d2eff8c9916fe2c4fe5c8d18158dbc977f662/src/lightning/fabric/plugins/environments/slurm.py#L106-L109,
see also
https://github.com/Lightning-AI/pytorch-lightning/blob/5a0d2eff8c9916fe2c4fe5c8d18158dbc977f662/src/lightning/fabric/plugins/environments/slurm.py#L159-L160, and https://github.com/Lightning-AI/pytorch-lightning/blob/5a0d2eff8c9916fe2c4fe5c8d18158dbc977f662/src/lightning/fabric/plugins/environments/slurm.py#L231-L232

Reply:
Got it. Thanks for the detailed info.